File: CocktailDebacle.API/Cocktail.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;


public class Cocktail
{
    [Key]
    public int Id { get; set; }

    [Required]
    public CocktailApiDrink Drink { get; set; } = new();

    [ForeignKey("CreatedByUserId")]
    public User? CreatedByUser { get; set; }
    public int? CreatedByUserId { get; set; }

    [InverseProperty("FavoriteCocktails")]
    public ICollection<User> FavoriteByUsers { get; set; } = new List<User>();
}

public class CocktailApiDrink
{
    [JsonPropertyName("IdDrink")]
    public string? IdDrink { get; set; }

    [JsonPropertyName("strDrink")]
    public string? StrDrink { get; set; }

    [JsonPropertyName("strGlass")]
    public string? StrGlass { get; set; }

    [JsonPropertyName("strCategory")]
    public string? StrCategory { get; set; }

    [JsonPropertyName("strDrinkThumb")]
    public string? StrDrinkThumb { get; set; }

    [JsonPropertyName("strInstructions")]
    public string? StrInstructions { get; set; }

    [JsonPropertyName("strIngredient1")]
    public string? StrIngredient1 { get; set; }

    [JsonPropertyName("strIngredient2")]
    public string? StrIngredient2 { get; set; }

    [JsonPropertyName("strIngredient3")]
    public string? StrIngredient3 { get; set; }

    [JsonPropertyName("strIngredient4")]
    public string? StrIngredient4 { get; set; }

    [JsonPropertyName("strIngredient5")]
    public string? StrIngredient5 { get; set; }

    [JsonPropertyName("strIngredient6")]
    public string? StrIngredient6 { get; set; }

    [JsonPropertyName("strMeasure1")]
    public string? StrMeasure1 { get; set; }

    [JsonPropertyName("strMeasure2")]
    public string? StrMeasure2 { get; set; }

    [JsonPropertyName("strMeasure3")]
    public string? StrMeasure3 { get; set; }

    [JsonPropertyName("strMeasure4")]
    public string? StrMeasure4 { get; set; }

    [JsonPropertyName("strMeasure5")]
    public string? StrMeasure5 { get; set; }

    [JsonPropertyName("strMeasure6")]
    public string? StrMeasure6 { get; set; }
}

File: frontend/src/app/models/cocktail.ts
export class Cocktail {
    Id: number;
    drink: CocktailApiDrink;
    CreatedByUserId?: number;
    FavoriteByUsers: string[]; // Array di ID o nomi degli utenti che hanno aggiunto il cocktail ai preferiti
  
    constructor(init?: Partial<Cocktail>) {
      this.Id = init?.Id || 0;
      this.drink = init?.drink || new CocktailApiDrink();
      this.CreatedByUserId = init?.CreatedByUserId;
      this.FavoriteByUsers = init?.FavoriteByUsers || [];
    }
  }
  
  export class CocktailApiDrink {
    IdDrink?: string;
    StrDrink?: string;
    StrDrinkAlternate?: string;
    StrTags?: string;
    StrVideo?: string;
    StrCategory?: string;
    StrIBA?: string;
    StrAlcoholic?: string;
    StrGlass?: string;
    StrInstructions?: string;
    StrInstructionsES?: string;
    StrInstructionsDE?: string;
    StrInstructionsFR?: string;
    StrInstructionsIT?: string;
    StrInstructionsZH_HANS?: string;
    StrInstructionsZH_HANT?: string;
    StrDrinkThumb?: string;
    StrIngredient1?: string;
    StrIngredient2?: string;
    StrIngredient3?: string;
    StrIngredient4?: string;
    StrIngredient5?: string;
    StrIngredient6?: string;
    StrIngredient7?: string;
    StrIngredient8?: string;
    StrIngredient9?: string;
    StrIngredient10?: string;
    StrIngredient11?: string;
    StrIngredient12?: string;
    StrIngredient13?: string;
    StrIngredient14?: string;
    StrIngredient15?: string;
    StrMeasure1?: string;
    StrMeasure2?: string;
    StrMeasure3?: string;
    StrMeasure4?: string;
    StrMeasure5?: string;
    StrMeasure6?: string;
    StrMeasure7?: string;
    StrMeasure8?: string;
    StrMeasure9?: string;
    StrMeasure10?: string;
    StrMeasure11?: string;
    StrMeasure12?: string;
    StrMeasure13?: string;
    StrMeasure14?: string;
    StrMeasure15?: string;
    StrImageSource?: string;
    StrImageAttribution?: string;
    StrCreativeCommonsConfirmed?: string;
    DateModified?: string;
  
    constructor(init?: Partial<CocktailApiDrink>) {
      Object.assign(this, init);
    }
  }